### PR TITLE
Food Models Fix

### PR DESCRIPTION
**Pull request by mirandawang**
Cherry picked the commits from FoodModels before merge conflicts happened while trying to interactive rebase Food Models branch. 

I set up the models for FoodItem and FoodCategory (renamed to just Item and Category), which both just have String names as fields. Category and Item have a One - to - Many relationship: Categories can have many Items, Item has one Category. I also created some basic Rspec tests for these Models, checking validations so name cannot be nil and all names must be unique. I added a helper method in Item model to downcase the names before saving so that the unique-ness would be case sensitive. Categories are hard coded so users cannot add or edit the categories. Items has a controller that handles create.

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Item < ApplicationRecord
+  belongs_to :category
+  validates_presence_of :name
+  validates_uniqueness_of :name
+
+  before_validation :downcase_name
+
+  def downcase_name
+    if self.name?
```
Comment (nhessler) [inline app/models/item.rb]: is there ever a time an Item could or would not have a name?

**Comment by purelogiq** [comment]
I like the way this PR is stuctured more but it looks like you lost some commits:

```
    Updating Gemfile.lock                                                   7fea40e
    Adding controller for Items.                                                    bb5ba93
    Fixing table names in database.                         f9875a5
    Fixing db schema and Gemfile.lock merge conflicts           7dc1d2d
```

Do you have a plan to get these changes back?

**Comment by nhessler** [comment]
made one small comment, but otherwise looks good! 👍

**Comment by nhessler** [comment]
@purelogiq, I think those were just merge conflict commits. these were caught as part of the individual cherry picked commits.

File: app/models/item.rb (hunk 1 above)
Comment (mirandawang) [inline app/models/item.rb]: Item should always have a name. There shouldn't be any items that are just blank. So I don't need that conditional!

Comment (mirandawang) [inline app/models/item.rb]: I included the conditional in order for a test to pass; I was testing that Item should not be valid without a name. However it just errors out (NoMethodError: undefined method 'downcase!' for nil:NilClass). Is that ok?

Comment (purelogiq) [inline app/models/item.rb]: I would change the `before_validation` to `before_save`. Validating an item doesn't depend on it's name being in downcase, all that matters is that it has a name. What matters is that when it is saved it's name is downcased.

This lets you remove the if condition.

Comment (mirandawang) [inline app/models/item.rb]: Will this still validate case-insensitive uniqueness though?

**Comment by purelogiq** [comment]
@mirandaaa It looks like you are missing the migrations that created the item and category tables. Right now you're depending on the `db/schema.rb` file to setup your database from scratch. Do you know where the migration files went? In general they should always be saved for situations where you need to update a database that can't just be recreated from scratch with a new schema.

If you need to get them back it looks like they are [here](https://github.com/mirandaaa/grocery/pull/2/commits/e5abff182a8c1cbaa4826da34c155d8ea1e7d76c). You'll need to copy the file names and contents exactly.

File: app/models/item.rb (hunk 1 above)
Comment (purelogiq) [inline app/models/item.rb]: @mirandaaa  You can use `validates_uniqueness_of :name, :case_sensitive => false` for that 😄

Comment (mirandawang) [inline app/models/item.rb]: It works! I had tried that earlier, but it appeared to not behave correctly which is why I did the downcase.

File: spec/models/category_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require "spec_helper"
+
+describe Category do
+  let(:dairy) { Category.new(name: "dairy") }
+  let(:items) { [Item.new(name: "milk", category: dairy), Item.new(name: "eggs", category: dairy)] }
```
Comment (purelogiq) [inline spec/models/category_spec.rb]: This line isn't being used anywhere, I guessing it's just here in preparation for more tests.
Line 17 in this file only checks for the association not for actual concrete items.

**Comment by purelogiq** [comment]
LGTM 👍  Since Nathan is good with it to I say you merge it in (via github) and rebase your other branch on top of the new master.
